projects/plugins/jetpack/extensions/blocks/subscriptions/class-jetpack-subscription-site.php

```diff
@@ -88,6 +109,60 @@ protected function get_post_end_placement_block_attributes( $default_attrs, $anc
 		return $default_attrs;
 	}
 
+	/**
+	 * Returns true if context is recognized as a header element.
+	 *
+	 * @param WP_Block_Template|WP_Post|array $context The block template, template part, or pattern the anchor block belongs to.
+	 *
+	 * @return bool
+	 */
+	protected function is_header_context( $context ) {
+		if ( $context instanceof WP_Post && $context->post_type === 'wp_navigation' ) {
+			return true;
+		}
+
+		if ( $context instanceof WP_Block_Template && $context->area === 'header' ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Handles Subscription block navigation placement.
+	 *
+	 * @return void
+	 */
+	protected function handle_subscribe_block_navigation_placement() {
+		global $wp_version;
+
+		$is_enabled = get_option( 'jetpack_subscriptions_subscribe_navigation_enabled', false );
+		if ( ! $is_enabled ) {
+			return;
+		}
+
+		if ( ! wp_is_block_theme() || version_compare( $wp_version, '6.5-beta2', '<' ) ) { // TODO Fallback for classic themes and wp core < 6.5-beta2.
+			return;
+		}
+
+		add_filter(
+			'hooked_block_types',
+			function ( $hooked_blocks, $relative_position, $anchor_block, $context ) {
+				if (
+					$anchor_block === 'core/navigation' &&
+					$relative_position === 'last_child' &&
+					self::is_header_context( $context )
+				) {
+					$hooked_blocks[] = 'jetpack/subscriptions';
```

Comment:
Somehow we need to add  `className` attribute to the block when injected, so that it appears in "button only" style.

Reply:
You need another filter to modify attributes, check this one:
https://github.com/Automattic/jetpack/blob/75b03102a816848ca9a6c0ca084887b6237a3f66/projects/plugins/jetpack/extensions/blocks/subscriptions/class-jetpack-subscription-site.php#L172-L174